src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getUser } from '../../Api/Api';
+import './CurrentUser.scss';
+
+export const CurrentUser = () => {
+  const dispatch = useDispatch();
+
+  const user = useSelector((state: RootState) => state.user);
+  const selectedUserId = useSelector((state: RootState) => state.selectedUserId);
+
+  useEffect(() => {
+    (async () => {
+      let userFromServer;
+      let serverError = false;
+
+      try {
+        userFromServer = await getUser(selectedUserId);
+      } catch (error) {
+        const str = 'An error has ocurred while getting data from server, please contact our support team';
+
+        serverError = true;
+        dispatch({ type: 'SET_USER_ERROR', payload: str });
+      }
+
+      if (!serverError) {
+        dispatch({ type: 'LOAD_USER', payload: userFromServer });
+      }
+    })();
+  }, [selectedUserId]);
+
+  const clearHandler = () => {
+    dispatch({ type: 'SET_SELECTED_USER_ID', payload: 0 });
+  };
+
+  const userLoader = () => {
+    const {
+      id, name, email, phone,
+    } = user;
+
+    if (id !== selectedUserId) {
```

Comment:
do you need selectedUserId?
you already have the user in redux store, use user.id